src/App.jsx

```diff
@@ -1,17 +1,33 @@
+import { useState} from 'react';
 import './App.css';
+import messagesData from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
+function App() {
+  const [entries, setEntries] = useState(messagesData);
+
+  const toggleLike = (id) => {
+    setEntries(entries => entries.map(entry => {
+      if (entry.id === id) {
+        return { ...entry, liked: !entry.liked};
+      } else {
+        return entry;
+      }
+    }));
+  };
 
-const App = () => {
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>{entries.filter(entry => entry.liked).length} ❤️s</h1>
```

Comment:
This isn't complex logic, but would still prefer to have it in a helper function. 

You can also use `reduce` to calculate the total likes:

```jsx
  // adding a bool to a number treats true as 1 and 0 as false
  const totalLikes = entries.reduce((totalLikes, entry) => (totalLikes + entry.liked), 0);
```